### ISSUE TITLE
Make --interactive aware of stdin

### ISSUE THREAD

**Issue by reedho** [borkdude/jet#31]
Currently `--interactive` bypass processing of stdin. It would be nice if we can start interactive session from e.g. clipboard as initial value: `pbpaste | jet --from json --interactive`.

I would love to provide patch if You think this is good.

Thank you.

**Comment by borkdude** [comment]
Since the interactive mode already reads user input from stdin, I'm not sure how this should work. Also, I'm considering deprecating the jet-lang in favor of https://github.com/borkdude/babashka.

**Comment by borkdude** [comment]
@reedho An alternative might be reading from a file explicitly with a `--file` argument. So:

``` 
jet --interactive --file foo.edn
```

**Comment by reedho** [comment]
Hm, yeah.. now i know that once stdin is given from the shell pipe, it is closed and can not be continued.

So, below is the hack, it read `/dev/tty`, tested ok with linux, and should also osx.
```
diff --git src/jet/jeti.clj src/jet/jeti.clj
index d70c976..dcbcffa 100644
--- src/jet/jeti.clj
+++ src/jet/jeti.clj
@@ -59,19 +59,19 @@
        (catch Exception e
          (println "Could not write to" (str file ":") (.getMessage e)))))
 
-(defn start-jeti! []
+(defn start-jeti! [init-val]
   (println "Welcome to jeti. The answer is just a few queries away!")
   (println "Running jet" (str "v" (str/trim (slurp (io/resource "JET_VERSION"))) "."))
   (println "Type :jeti/help to print help.")
   (println)
   (let [init-id (new-id nil)]
     (loop [{:keys [:bookmarks :print-level :print-length] :as state}
-           {init-id ::start
+           {init-id (or init-val ::start)
             :bookmarks []
             :print-length 5
             :print-level 5}
            previous-id nil
-           current-id nil]
+           current-id (when init-val init-id)]
       (let [current-val (get state current-id)
             prev-val (get state previous-id)
             same? (= current-id previous-id)
diff --git src/jet/main.clj src/jet/main.clj
index 6d7263a..4cbec92 100644
--- src/jet/main.clj
+++ src/jet/main.clj
@@ -51,9 +51,8 @@
   (let [{:keys [:from :to :keywordize
                 :pretty :version :query
                 :interactive]} (parse-opts args)]
-    (cond version
-      (println (str/trim (slurp (io/resource "JET_VERSION"))))
-      interactive (start-jeti!)
+    (cond
+      version (println (str/trim (slurp (io/resource "JET_VERSION"))))
       :else
       (let [in (slurp *in*)
             input (case from
@@ -62,10 +61,15 @@
                     :transit (formats/parse-transit in))
             input (if query (q/query input query)
                       input)]
-        (case to
-          :edn (println (formats/generate-edn input pretty))
-          :json (println (formats/generate-json input pretty))
-          :transit (println (formats/generate-transit input)))))))
+        (cond
+          interactive (binding [*in* (clojure.lang.LineNumberingPushbackReader.
+                                      (clojure.java.io/reader "/dev/tty"))]
+                        (start-jeti! input))
+          :else
+          (case to
+            :edn (println (formats/generate-edn input pretty))
+            :json (println (formats/generate-json input pretty))
+            :transit (println (formats/generate-transit input))))))))
 
 ;;;; Scratch
```

**Comment by borkdude** [comment]
@reedho Maybe you can submit this as a pull request, to see if the tests pass? Also I find that easier to read.

**Comment by reedho** [comment]
Done, please check.

**Comment by reedho** [comment]
As of e617e4c, some code has been merged (thank you!) to cover interactive usage as follow:

`jet --interactive` - Basic use, drop to repl without initial value.

`jet --interactive '{:init-val 10}'` - Enter repl with init-val 10.

`jet --interactive '10'` - Same as above.

`jet --interactive '{"init-val": {"a": 100}}' --from json --keywordize` or simply `jet --interactive '{"a": 100}' --from json --keywordize` - Provide init-val as json.

But it still can not handle shell pipe as normal cli do:

`echo '{:a 100}' | jet --interactive` or the similar like `pbpaste | jq .SomeField | jet --interactive --from json`

Work around is something like:

`jet --interactive "$(cat data.json)" --from json` - For bash, work even if data.json is single or multi/pretty printed.

`jet --interactive (cat data.json) --from json` - For fish, ONLY work if data.json is not multi line contents.

There is still option to seamlessly make shell piping work from information described here: https://stackoverflow.com/a/23419451

**Comment by borkdude** [comment]
I have changed the behavior slightly. The `--interactive` command now takes the first command for the shell as an argument. Passing in literal values still works, but you can now also read files using `:jeti/slurp`. This is now released in v0.0.10.
